src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1758,13 +1772,98 @@ private boolean evaluateNext() {
         return false;
     }
 
+    @Override
+    public void transcodeTo(MacroAwareIonWriter writer) throws IOException {
+        registerIvmNotificationConsumer((x, y) -> {
+            resetEncodingContext();
+            writer.startEncodingSegmentWithIonVersionMarker();
+        });
+        while (transcodeNextTo(writer) != Event.NEEDS_DATA);
+    }
+
+    /**
+     * Transcodes the next value, and any encoding directives that may precede it,
+     * to the given writer.
+     * @param writer the writer to which the value will be transcoded.
+     * @return the result of the operation.
+     * @throws IOException if thrown during writing.
+     */
+    Event transcodeNextTo(MacroAwareIonWriter writer) throws IOException {
```

Comment:
I found this method (and `nextValue()`) difficult to follow.

Reply:
Agreed, I'm going to think more about about a better factoring. Opened https://github.com/amazon-ion/ion-java/issues/1004